packages/front-end/components/Dimensions/DimensionForm.tsx

```diff
@@ -31,101 +33,121 @@ const DimensionForm: FC<{
+
   const dsProps = dsObj?.properties;
-  const sql = dsProps?.queryLanguage === "sql";
+  const supportsSQL = dsProps?.queryLanguage === "sql";
+
+  const sql = form.watch("sql");
 
   const requiredColumns = useMemo(() => {
     return new Set([userIdType, "value"]);
   }, [userIdType]);
 
   return (
-    <Modal
-      close={close}
-      open={true}
-      header={current.id ? "Edit Dimension" : "New Dimension"}
-      submit={form.handleSubmit(async (value) => {
-        if (sql) {
-          validateSQL(value.sql, [value.userIdType, "value"]);
-        }
-
-        await apiCall(
-          current.id ? `/dimensions/${current.id}` : `/dimensions`,
-          {
-            method: current.id ? "PUT" : "POST",
-            body: JSON.stringify(value),
-          }
-        );
-        mutateDefinitions();
-      })}
-    >
-      <Field label="Name" required {...form.register("name")} />
-      <Field
-        label="Owner"
-        options={memberUsernameOptions}
-        comboBox
-        {...form.register("owner")}
-      />
-      <SelectField
-        label="Data Source"
-        required
-        value={form.watch("datasource")}
-        onChange={(v) => form.setValue("datasource", v)}
-        placeholder="Choose one..."
-        options={datasources.map((d) => ({
-          value: d.id,
-          label: `${d.name}${d.description ? ` — ${d.description}` : ""}`,
-        }))}
-        className="portal-overflow-ellipsis"
-      />
-      {dsProps.userIds && (
-        <SelectField
-          label="Identifier Type"
-          required
-          value={userIdType}
-          onChange={(v) => form.setValue("userIdType", v)}
-          options={(dsObj.settings.userIdTypes || []).map((t) => {
-            return {
-              label: t.userIdType,
-              value: t.userIdType,
-            };
-          })}
+    <>
+      {sqlOpen && datasource && (
+        <EditSqlModal
+          close={() => setSqlOpen(false)}
+          datasourceId={dsObj.id || ""}
+          placeholder={`SELECT\n      ${userIdType}, date\nFROM\n      mytable`}
+          requiredColumns={Array.from(requiredColumns)}
+          value={sql}
+          save={async (sql) => form.setValue("sql", sql)}
         />
       )}
-      {sql ? (
-        <SQLInputField
-          userEnteredQuery={form.watch("sql")}
-          datasourceId={dsObj.id}
-          form={form}
-          requiredColumns={requiredColumns}
-          placeholder={`SELECT\n      ${userIdType}, browser as value\nFROM\n      users`}
-          helpText={
-            <>
-              Select two columns named <code>{userIdType}</code> and{" "}
-              <code>value</code>
-            </>
+      <Modal
+        close={close}
+        open={true}
+        size="md"
+        header={current.id ? "Edit Dimension" : "New Dimension"}
+        submit={form.handleSubmit(async (value) => {
+          if (supportsSQL) {
+            if (!sql) throw new Error("SQL cannot be empty");
```

Comment:
Should use `value.sql`, not `sql` here.  We can also move this check to `validateSQL` if we want since I think it's going to repeated everywhere we use it.

Reply:
I removed it as the check also exists within `validateSQL`. Also removed it from `SegmentForm`.